### ISSUE TITLE
[Tests] nobody user does not have permissions to open /etc/passwd

### ISSUE THREAD

**Issue by vamsikrishna935** [lsds/sgx-lkl#224]
Identified issue while fixing setfsuid03 testcase. 
In setfsuid03 test case at line no 55 : https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L55
getpwuid(uid) function is being called with nobody user.

From here it goes in the below mentioned pattern and reaches __getpw_a function at src/passwd/getpw_a.c:24 and tries to open /etc/passwd with **nobody** user. Here it fails with Permission denied error (EACCESS: 13).

```
**Code flow:**
getpwuid (uid=1) at src/passwd/getpwent.c:28
28              __getpw_a(0, uid, &pw, &line, &size, &res);
(gdb) s
__getpw_a (name=0x0, uid=1, pw=0x7fff00ad12c0 <pw>, buf=0x7fff00b1aa50 <line>, size=0x7fff00b1aa58 <size>,
    res=0x7fff40b76d98) at src/passwd/getpw_a.c:24

int __getpw_a(const char *name, uid_t uid, struct passwd *pw, char **buf, size_t *size, struct passwd **res)
21      {
22              FILE *f;
23              int cs;
24              int rv = 0;
25
26              *res = 0;
27
28              pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
		f = fopen("/etc/passwd", "rbe");
31              if (!f) {
32                      rv = errno;
```

**Comment by vamsikrishna935** [comment]
@prp found an issue.

**Comment by hukoyu** [comment]
@prp  @vamsikrishna935  provided code link, detailed problem description, gdb output, parameters. Please let me know if you need anything else.

cc @hukoyu

**Comment by prp** [comment]
Thanks. Can you:
- update the comment to use proper markdown formatting? For example, code or debugging snippets should be escaped with back-ticks.
- include the permissions of `/etc/passwd` by running `ls -la /etc/passwd` inside the enclave?
- double-check what the uid is that the programs executes with?

I would like to understand if the permissions denied error is correct or not.

**Comment by vamsikrishna935** [comment]
@prp Updated the code format.

Permissions for /etc/passwd: 
```
[[  SGX-LKL ]] libc_start_main_stage2(): Calling app main: /bin/ls
-rw-r--r--    1 root     root          1224 Jun  7  2018 /etc/passwd
```

The test is running with nobody user: Please find below reference.
https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L23
https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L41
https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L55
https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L90
https://github.com/linux-test-project/ltp/blob/master/testcases/kernel/syscalls/setfsuid/setfsuid03.c#L95

**Comment by prp** [comment]
The permissions of `/etc/passwd` look fine.

Can you check that it is indeed possible to run as user 'nobody'?

Can you write a program that opens and reads `/etc/passwd` successfully? Can you make that program run as user 'nobody'?

**Comment by vamsikrishna935** [comment]
> 
> 
> The permissions of `/etc/passwd` look fine.
> 
> Can you check that it is indeed possible to run as user 'nobody'?
> 
> Can you write a program that opens and reads `/etc/passwd` successfully? Can you make that program run as user 'nobody'?

@prp Sure peter

**Comment by SeanTAllen** [comment]
This needs investigation to be picked up.

**Comment by SeanTAllen** [comment]
This might be the same issue as #236. I'm checking now.

**Comment by SeanTAllen** [comment]
This is fixed. It was fixed by #723.

**Comment by SeanTAllen** [comment]
PR opened to close this issue: https://github.com/lsds/sgx-lkl/pull/744
